proto/cosmos/crypto/ed25519/keys.proto

```diff
@@ -11,13 +12,28 @@ option go_package = "github.com/cosmos/cosmos-sdk/crypto/keys/ed25519";
 // ADR-28. Nevertheless, you will like to use ed25519 in app user level
 // then you must create a new proto message and follow ADR-28 for Address construction.
 message PubKey {
-  option (gogoproto.goproto_stringer) = false;
+  option (cosmos.msg.v1.legacy_amino_name) = "tendermint/PubKeyEd25519";
+  // The Amino encoding is simply the inner bytes field, and not the Amino
+  // encoding of the whole PubKey struct.
+  //
+  // Example (JSON):
+  // s := PubKey{Key: []byte{0x01}}
+  // out := AminoJSONEncoder(s)
+  //
+  // Then we have:
+  // out == `"MQ=="`
+  // out != `{"key":"MQ=="}`
+  option (cosmos.msg.v1.legacy_amino_encoding) = "bytes field";
```

Comment:
Example of using `cosmos.msg.v1.legacy_amino_encoding`

Reply:
Any ideas what to put instead of `bytes field` here?

Reply:
wouldn't `legacy_amino_encoding` be for the field itself, not the message?

In general, I would use `legacy_amino_encoding` to place hints per field for how to generate the encoders/decoders. 

I would suggest `legacy_amino_encoding` isn't needed at the message level, only field level.

Reply:
Yes, initially, `legacy_amino_encoding` was on the field (via scalars and ScalarDescriptor). But the `pubkey` case you pointed below made me thinkg of adding a message-level option too.
https://github.com/cosmos/cosmos-sdk/blob/9678f313ad145aef2305a2f23e40171aabdea44f/proto/cosmos/staking/v1beta1/tx.proto#L70

This any can contain either secp256k1, ed25519 or multisig pubkeys. We might support new pubkeys too. Which `legacy_amino_encoding` string would cater for all pubkey types?

See encoding of current pubkeys here: https://github.com/amaurym/go-amino-spec/blob/main/spec_test.go#L36-L51

Reply:
> Any ideas what to put instead of bytes field here?

`bytes_field` sounds strange. I would use `bytes`, `base64`, etc...